### ISSUE TITLE
Does not work with new features in solidity 0.8.18 (named mapping parameters)

### ISSUE THREAD

**Issue by cyberhorsey** [sc-forks/solidity-coverage#782]
Running coverage with solc ^0.8.18 gives:
```ts
 Could not instrument: input 'addr' expecting '=>'
```

This may be an issue with the `solidity-parser` plugin though?

**Comment by cyberhorsey** [comment]
I opened a PR for this.
https://github.com/sc-forks/solidity-coverage/pull/783

**Comment by eladiosch** [comment]
Is there an ETA for this? The fix looks good, so please merge so we can continue to use this tool with the new version of Solidity

**Comment by HardlyDifficult** [comment]
Workaround: if you are using `yarn`, in your package.json add this and then run `yarn` again.

```json
  "resolutions": {
    "@solidity-parser/parser": "0.16.0"
  },
```

I believe `npm` can do the same with `overrides` but have not tested that.

**Comment by cgewecke** [comment]
Resolved in 0.8.4
